### DISCUSSION TITLE
OCI based registry authentication for helm dependencies

### DISCUSSION THREAD

**Discussion by ankitabhopatkar13** [renovatebot/renovate#11607] [Request Help] [answered]
I am using renovate-runner(3.17.0) on gitlab-ci. My helm charts are hosted in a private Gitlab Container Registry which is basically OCI based.

I was wondering if I can use the helm charts hosted in private Gitlab Container Registry. I saw the documentation [here](https://docs.renovatebot.com/docker/), tried to use the hostRules but I still get the following error -

```
{
                     "topic": "dependency-name",
                     "message": "Failed to look up dependency dependency-name"
                   }
```
I am trying to authenticate with Gitlab deploy token and the configuration looks like below -

```
  hostRules: 
    {
      hostType: 'docker',
      matchHost: process.env.CI_REGISTRY,
      username: 'ci-deploy-token',
      password: process.env.GITLAB_CI_CHARTS_DEPLOY_TOKEN,
    },
```

I also tried with token, but same error. 

On local machine I run `helm registry login` [command](https://helm.sh/docs/topics/registries/#login) before downloading the dependency.

Is there any other way I can authenticate to OCI registry for helm charts?

Thanks in Advance!

**Comment by viceice** [comment]
We need more information on this. Are you able to create a minimal reproduction repo?

https://github.com/renovatebot/renovate/blob/HEAD/docs/development/minimal-reproductions.md

**Comment by ankitabhopatkar13** [comment]
@viceice 

I created a small repo with the README explaining the problem -

https://github.com/ankitabhopatkar13/renovate-helm-oci

Let me know if I can provide more details!

**Comment by pspurlock07** [comment]
@viceice @ankitabhopatkar13 Did you ever get an answer on this? We are running into the exact same issue.

**Comment by viceice** [comment]
Did you tried to override to use the docker datasource? I'm pretty sure our helm datasource can handle OCI registries.

Did anybody tried with public OCI registry? Did it work without credentials?

**Reply by pspurlock07** [reply]
I did, this is my current config.js:

```javascript
  module.exports = {
    endpoint: process.env.CI_API_V4_URL,
      {
        matchHost: 'reg.gitlab.com/ace/chart-gallery/deployment/charts',
        hostType: 'docker',
        username: 'other-user',
        password: process.env.PASS2,
      },
```

I also tried variations of matchHost with `https://` and `oci://` and it's same error always:

`DEBUG: Datasource 404 (repository=devsecops/testapp/test-app-chart)
       "datasource": "docker",
       "lookupName": "reg.gitlab.com/ace/chart-gallery/deployment/charts",
       "url": "https://reg.gitlab.com/v2/ace/chart-gallery/deployment/charts/tags/list?n=10000"`
       
     

I did not try a public OCI reg, but I also don't know of any to try.

**Reply by viceice** [reply]
So we need somebody to fiddle out the OCI API specs and then we can extend the helm datasource to support this.

When sombody has collected all infos, please check for existing issues and fill the infos / or cretae a new issue.

**Reply by HonkingGoose** [reply]
Is this issue maybe related?

- https://github.com/renovatebot/renovate/issues/10659

**Reply by pspurlock07** [reply]
ahh, yes! Sorry I missed that one. Ok, it sounds like there are 2 issues going on. I created this issue about this at the advice of vice: https://github.com/renovatebot/renovate/issues/13524

Mine points to the url path seems wrong, so I don't see how it would work privately or otherwise.

The one you linked is more toward authenticating to a private repo.

**Reply by HonkingGoose** [reply]
> ahh, yes! Sorry I missed that one.

No problem, we got there in the end. 😉

**Comment by ankitabhopatkar13** [comment, marked as answer]
After [this PR](https://github.com/renovatebot/renovate/pull/13726) was merged, this problem was solved.

This question can be marked as `Answered`
